src/identity/keystore.rs

```diff
@@ -205,15 +420,39 @@ impl Keystore {
         transcript_items: impl Iterator<Item = (&'static [u8], either::Either<&'a [u8], u64>)> + 'a,
     ) -> impl core::future::Future<Output = Result<VrfSignature, SignVrfError>> + 'a {
         async move {
-            let guarded = self.guarded.lock().await;
+            let mut guarded = self.guarded.lock().await;
             let key = guarded
                 .keys
                 .get(&(key_namespace, *public_key))
                 .ok_or(SignVrfError::Sign(SignError::UnknownPublicKey))?;
 
             match key {
-                PrivateKey::MemoryEd25519(_) => Err(SignVrfError::WrongKeyAlgorithm),
-                PrivateKey::MemorySr25519(key) => {
+                PrivateKey::MemoryEd25519(_) | PrivateKey::FileEd25519 => {
+                    Err(SignVrfError::WrongKeyAlgorithm)
+                }
+                PrivateKey::MemorySr25519(_) | PrivateKey::FileSr25519 => {
+                    let key = match key {
+                        PrivateKey::MemorySr25519(key) => Cow::Borrowed(key),
+                        PrivateKey::FileSr25519 => {
+                            match Self::load_sr25519_from_file(
+                                self.path_of_key_sr25519(key_namespace, public_key).unwrap(),
+                            )
+                            .await
+                            {
+                                Ok(key) => {
+                                    drop(guarded);
+                                    // TODO: is creating the signing context expensive?
```

Comment:
seems unrelated

Reply:
Ah right, I copy-pasted some code and this comment moved at the wrong place